x-pack/plugins/security_solution/public/management/pages/endpoint_hosts/view/details/components/activity_log_date_range_picker/index.tsx

```diff
@@ -0,0 +1,135 @@
+
+const DatePickerWrapper = styled.div`
+  width: ${(props) => props.theme.eui.fractions.single.percentage};
+  background: white;
+`;
+const StickyFlexItem = styled(EuiFlexItem)`
+  position: sticky;
+  top: ${(props) => props.theme.eui.euiSizeM};
+  z-index: 1;
+`;
+
+export const DateRangePicker = memo(() => {
+  const dispatch = useDispatch();
+  const { page, pageSize, startDate, endDate } = useEndpointSelector(getActivityLogDataPaging);
+
+  const isInvalidDateRange = useMemo(
+    () => (startDate && endDate ? moment(startDate) > moment(endDate) : false),
+    [startDate, endDate]
+  );
+
+  const onClear = useCallback(
+    ({ clearStart = false, clearEnd = false }: { clearStart?: boolean; clearEnd?: boolean }) => {
+      dispatch({
+        type: 'endpointDetailsActivityLogUpdatePaging',
+        payload: {
+          disabled: false,
+          page,
+          pageSize,
+          startDate: clearStart ? undefined : startDate,
+          endDate: clearEnd ? undefined : endDate,
+        },
+      });
+    },
+    [dispatch, endDate, startDate, page, pageSize]
+  );
+
+  const onChangeStartDate = useCallback(
+    (date) => {
+      if (isInvalidDateRange) {
+        return;
+      }
+      dispatch({
+        type: 'endpointDetailsActivityLogUpdatePaging',
+        payload: {
+          disabled: false,
+          page,
+          pageSize,
+          startDate: date ? date?.toISOString() : undefined,
+          endDate: endDate ? endDate : undefined,
+        },
+      });
+    },
+    [dispatch, isInvalidDateRange, endDate, page, pageSize]
+  );
+
+  const onChangeEndDate = useCallback(
+    (date) => {
+      if (isInvalidDateRange) {
+        return;
+      }
+      dispatch({
+        type: 'endpointDetailsActivityLogUpdatePaging',
+        payload: {
+          disabled: false,
+          page,
+          pageSize,
+          startDate: startDate ? startDate : undefined,
+          endDate: date ? date.toISOString() : undefined,
+        },
+      });
+    },
+    [dispatch, isInvalidDateRange, startDate, page, pageSize]
+  );
+
+  return (
+    <StickyFlexItem grow={false}>
+      <EuiFlexGroup justifyContent="flexEnd" responsive>
+        <DatePickerWrapper>
+          <EuiFlexItem>
+            <EuiDatePickerRange
+              data-test-subj="activityLogDateRangePicker"
+              startDateControl={
+                <EuiDatePicker
+                  aria-label="Start date"
+                  endDate={endDate ? moment(endDate) : undefined}
+                  isInvalid={isInvalidDateRange}
+                  maxDate={moment(startDate) || moment()}
+                  onChange={onChangeStartDate}
+                  onClear={() => onClear({ clearStart: true })}
+                  placeholderText={i18.ACTIVITY_LOG.datePicker.startDate}
+                  selected={startDate ? moment(startDate) : undefined}
+                  showTimeSelect
+                  startDate={startDate ? moment(startDate) : undefined}
```

Comment:
I'm wondering if the default date should be now instead of undefined. Is this a desired scenario?

Reply:
Yes, it is desired. The idea is not to have any dates selected in the beginning. Essentially the filter is not active when we load the activity log tab.